src/NuGet.Core/NuGet.Common/PathUtil/PathUtility.cs

```diff
@@ -69,5 +70,64 @@ private static string EnsureTrailingCharacter(string path, char trailingCharacte
 
             return path + trailingCharacter;
         }
+
+        public static bool IsFileSystemCaseInsensitive
+        {
+            get { return _isFileSystemCaseInsensitive.Value; }
+        }
+
+        private static bool CheckIfFileSystemIsCaseInsensitive()
+        {
+            if (RuntimeEnvironmentHelper.IsWindows)
+            {
+                return true;
+            }
+            else
+            {
+                var listOfPathsToCheck = new string[]
+                {
+                    NuGetEnvironment.GetFolderPath(NuGetFolderPath.NuGetHome),
+                    NuGetEnvironment.GetFolderPath(NuGetFolderPath.Temp),
+                    Directory.GetCurrentDirectory()
+                };
+
+                var isCaseInsensitive = true;
+                foreach (var path in listOfPathsToCheck)
+                {
+                    bool ignore;
+                    var result = CheckCaseSenstivityRecursivelyTillDirectoryExists(path, out ignore);
+                    isCaseInsensitive = ignore ? isCaseInsensitive : isCaseInsensitive && result;
+                }
+                return isCaseInsensitive;
+            }
+        }
+
+        private static bool CheckCaseSenstivityRecursivelyTillDirectoryExists(string path, out bool ignoreResult)
+        {
+            var parentDirectoryFound = true;
+            ignoreResult = true;
+            while (true)
+            {
+                if (path.Length <= 1)
+                {
+                    ignoreResult = true;
+                    parentDirectoryFound = false;
+                    break;
+                }
+                if (Directory.Exists(path))
+                {
+                    ignoreResult = false;
+                    break;
+                }
+                path = Path.GetDirectoryName(path);
+            }
+
+            if (parentDirectoryFound)
+            {
+                return Directory.Exists(path.ToLower()) && Directory.Exists(path.ToUpper());
```

Comment:
ToLowerInvariant, ToUpperInvariant maybe? I'm wondering about the turkish I problem here. it could probably only happen from using the current directory here